File: starterkits/java/src/main/java/codes/blitz/game/message/game/GameMessage.java
package codes.blitz.game.message.game;

import java.util.List;
import java.util.Map;
import java.util.function.Function;
import java.util.stream.Collectors;

public class GameMessage {
	private int tick;
	private int totalTick;
	private String crewId;
	private List<Crew> crews;
	private GameMap map;
	private Map<String, Crew> crewsMapById;
	private Rules rules;

	public int getTick() {
		return tick;
	}

	public int getTotalTick() {
		return totalTick;
	}

	public String getCrewId() {
		return crewId;
	}

	public List<Crew> getCrews() {
		return crews;
	}

	public GameMap getGameMap() {
		return map;
	}

	public Map<String, Crew> getCrewsMapById() {
		if (crewsMapById == null) {
			crewsMapById = crews.stream().collect(
					Collectors.toMap(Crew::getId, Function.identity()));
		}
		return crewsMapById;
	}

	public Rules getRules() {
		return rules;
	}
}

File: starterkits/csharp/game/GameMessage.cs
using System.Collections.Generic;
using System.Linq;
using Blitz2021;

namespace Blitz2020
{
    public class Rules {
        public int MAX_MINER_CARGO;
        public int MAX_CART_CARGO;
        public int MAX_MINER_MOVE_CARGO;
    }

    public class GameMessage
    {
        public int tick;

        public int totalTick;

        public string crewId;
        public List<Crew> crews;

        public Map map;

        public Rules rules;

        public Dictionary<string, Crew> getCrewsMapById
        {
            get { return this.crews.ToDictionary(p => p.id, p => p); }
        }
    }
}